### PR TITLE
Create a basic mailchimp client

### PR DESCRIPTION
**Pull request by TickleThePanda**
`mailing-list-client` - This project provides an API for integrating with a mailing list service. It provides an implementation of the API for Mailchimp.

`mailchimp-webhooks` - This project provides an small example Spring Boot client that can listen to webhooks from Mailchimp that register user details changing in Mailchimp.

Things it doesn't address yet:
 - #2 - retrieving a list of people when it exceeds the limit (Mailchimp paging)

### CODE REVIEW

File: mailing-list-client/src/main/java/uk/ac/stfc/facilities/mailing/mailchimp/MailchimpClient.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+     */
+    public static MailchimpClient getInstance(MailchimpClientConfiguration configuration) {
+        return new MailchimpClient(MailchimpInternalHttpClient.getInstance(configuration));
+    }
+
+    private static String convertEmailToId(String email) {
+        return DigestUtils.md5Hex(email.toLowerCase());
+    }
+
+    private final MailchimpInternalHttpClient httpClient;
```
[the remark this answers is not in the document]
Comment (TickleThePanda) [reply mailing-list-client/src/main/java/uk/ac/stfc/facilities/mailing/mailchimp/MailchimpClient.java]: What about moving the `convertEmailToId` to the top instead, I think that improves the readability more?

File: mailing-list-client/src/main/java/uk/ac/stfc/facilities/mailing/mailchimp/MailchimpInternalHttpClient.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+    private final CloseableHttpClient client;
+    private final HttpContext context;
+    private final Gson gson = new GsonBuilder()
+            .registerTypeAdapter(LocalDateTime.class, new LocalDateTimeAdapter())
+            .create();
+
+    private final MailchimpClientConfiguration configuration;
+
+    MailchimpInternalHttpClient(
```
Comment (TickleThePanda) [inline mailing-list-client/src/main/java/uk/ac/stfc/facilities/mailing/mailchimp/MailchimpInternalHttpClient.java]: Comment warning that this should not be used.

Comment (TickleThePanda) [reply mailing-list-client/src/main/java/uk/ac/stfc/facilities/mailing/mailchimp/MailchimpInternalHttpClient.java]: Deprecated annotation

File: mailing-list-client/src/test/java/uk/ac/stfc/facilities/mailing/mailchimp/MailchimpInternalHttpClientTest.java [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+        }
+
+        @Override
+        public int hashCode() {
+            return value != null ? value.hashCode() : 0;
+        }
+    }
+
+    // dash is required in the api key
+    private static final MailchimpClientConfiguration config = new MailchimpClientConfiguration("abcd-ab1");
```
Comment (TickleThePanda) [inline mailing-list-client/src/test/java/uk/ac/stfc/facilities/mailing/mailchimp/MailchimpInternalHttpClientTest.java]: Check variable name

File: mailing-list-client/src/test/java/uk/ac/stfc/facilities/mailing/mailchimp/MailchimpClientIT.java [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+
+                    @Test
+                    @DisplayName("members from a list")
+                    public void deleteMemberFromList() throws MailingListClientException {
+                        mailchimpClient.deleteMemberFromList(listId, expectedUserEmail);
+                    }
+                }
+
+                @Nested
+                @DisplayName("a non-existant member")
```
Comment (srconway) [inline mailing-list-client/src/test/java/uk/ac/stfc/facilities/mailing/mailchimp/MailchimpClientIT.java]: non-existent

**Pull request merged**